pandas/core/series.py

```diff
@@ -4447,6 +4447,35 @@ def fillna(
             downcast=downcast,
         )
 
+    @Appender(
+        """
+        Parameters
+        ----------
+        item : int
+            Index of the element that needs to be removed.
+
+        Returns
+        -------
+        Value that is popped from series.
+
+        Examples
+        --------
+        >>> ser = pd.Series([1,2,3])
+
+        >>> ser.pop(0)
+        1
+
+        >>> ser
+        1    2
+        2    3
+        dtype: int64
+        """
+    )
+    @Substitution(klass="Series")
+    @Appender(generic._shared_docs["pop"])
+    def pop(self, item):
```

Comment:
This will need the same annotation as `pop` in the parent class

Reply:
I think would be just
```suggestion
    def pop(self, item: Label) -> Any:
```

Reply:
and for DataFrame i think this is appropriate. 
`def pop(self, item) -> Series:`

Reply:
@2796gaurav have you tried running `mypy` on that? I think it requires that the base class and subclasses all follow the same signature

Reply:
My bad, seems mypy allows it. So, can you try Simon's suggestion for `Series.pop`, and `def pop(self, item: Label) -> Series:` for `DataFrame.pop`?

Reply:
sure. sounds good